server/src/com/thoughtworks/go/server/service/UserService.java

```diff
@@ -131,6 +120,40 @@ private boolean userNameListContainsAdmin(List<String> enabledUserNames) {
         return false;
     }
 
+    public void update(User user, TriState enabled, TriState emailMe, String email, String checkinAliases, LocalizedOperationResult result) {
+        if (enabled.isTrue()) {
+            enable(Arrays.asList(user.getName()), result);
+        }
+
+        if (enabled.isFalse()) {
+            disable(Arrays.asList(user.getName()), result);
+        }
+
+        user = findUserByName(user.getName());
```

Comment:
Prefer us not doing this as the controller has loaded the user.
Or change the method to take in the username and then do the load.